packages/fuels-types/src/param_types.rs

```diff
@@ -388,6 +385,18 @@ fn try_vector(the_type: &Type) -> Result<Option<ParamType>> {
     Ok(Some(ParamType::Vector(Box::new(vec_elem_type))))
 }
 
+fn try_bytes(the_type: &Type) -> Result<Option<ParamType>> {
+    if !["struct std::bytes::Bytes", "struct Bytes"].contains(&the_type.type_field.as_str()) {
```

Comment:
Here, I don't think we should accept `struct Bytes` as a correct option. The `Bytes` type in Sway canonically means `std::bytes::Bytes`, I do think it would be confusing for the user to be able to declare its own `Bytes` type as a mapping to the `std::bytes::Bytes` type, it's a footgun. That's if I understand correctly of course.

Reply:
We have to accept `Bytes` since that is the only way `std::bytes::Bytes` are going to show up without the new `--json-call-paths` (or something similar) feature flag. 

This is the same way we've done it for vectors before. It does mean that if the user has a `Bytes` structure it will match. But then again if the user had a `Vec` structure it would have matched before.

One way out of that predicament is if we match the full declaration of the type, but that would make this code a bit more complicated.

Another way is to just wait until the call paths thing becomes the default.

Reply:
Ok so just to make sure I understand : this is temporary until the `--json-call-paths` flag becomes the default? Then we would not match on this?

Reply:
Once the call paths become default we'll always match on the full `std::bytes::Bytes` path. 

But, as I type this, a refactor is underway by @hal3e which will remove all these `try_*` helpers. So it might end up being removed even sooner.